lib/logstash/inputs/couchdb_changes.rb

```diff
@@ -181,7 +190,8 @@ def build_uri
 
   private
   def reconnect?
-    sleep(@always_reconnect ? @reconnect_delay : 0)
+    # Loop each second so stop? can work its magic
+    @reconnect_delay.times { sleep(1) unless stop? } if @always_reconnect
```

Comment:
Have you considered replacing this with the new stud `stoppable_sleep` method? I don't believe it was out at the time this code was written, it might be nice to have it here. There's an [example here](https://github.com/logstash-plugins/logstash-mixin-rabbitmq_connection/blob/master/lib/logstash/plugin_mixins/rabbitmq_connection.rb#L138).